src/server/editorServices.ts

```diff
@@ -1372,24 +1372,50 @@ namespace ts.server {
             this.projectToSizeMap.forEach(val => (availableSpace -= (val || 0)));
 
             let totalNonTsFileSize = 0;
+
             for (const f of fileNames) {
                 const fileName = propertyReader.getFileName(f);
                 if (hasTypeScriptFileExtension(fileName)) {
                     continue;
                 }
+
                 totalNonTsFileSize += this.host.getFileSize(fileName);
+
                 if (totalNonTsFileSize > maxProgramSizeForNonTsFiles) {
+                    logExceedLimit.call(this);
                     // Keep the size as zero since it's disabled
                     return true;
                 }
             }
 
             if (totalNonTsFileSize > availableSpace) {
+                logExceedLimit.call(this);
                 return true;
             }
 
             this.projectToSizeMap.set(name, totalNonTsFileSize);
             return false;
+
+            function logExceedLimit(this: ProjectService) {
+                const files: { name: string, size: number }[] = [];
+                let totalNonTsFileSize = 0;
+                for (const f of fileNames) {
+                    const name = propertyReader.getFileName(f);
+                    if (hasTypeScriptFileExtension(name)) {
+                        continue;
+                    }
+
+                    const size = this.host.getFileSize(name);
+                    totalNonTsFileSize += size;

```

Comment:
do you need to compute this again?

Reply:
consider:

```ts
fileNames
   .map(f => hasTypeScriptFileExtension(name) ? 0 : { name, size: host.getFileSize(name) })
   .sort((a, b) => b.size - a.size)
   .slice(0,5)
   .map(f => `${f.name} (${f.size})`)
   .join(",")
```

Reply:
That's a good idea, but I'll separate formatting from the chain.